feat(bitbucket): add API token authentication support and deprecate A…

sanjeevpenupala opened the pull request:
# Bitbucket API Token Authentication Support

## Summary

This PR adds support for Bitbucket API tokens as an authentication method for the Bitbucket plugin, in response to Atlassian's deprecation timeline for App passwords (creation discontinued September 9, 2025; all deactivated June 9, 2026).

**Key Changes:**
- **Backend**: Added `UsesApiToken` boolean field to track authentication type (both use HTTP Basic Auth with `username:credential` format)
- **Frontend**: Custom authentication component with radio selection between "API Token (Recommended)" and "App Password (Deprecated)"
- **Migration**: Automatic backward compatibility for existing App password connections (`usesApiToken = false` by default)
- **UX**: Deprecation warnings in UI and logs, dynamic username guidance (email for API tokens vs. username for App passwords)
- **Documentation**: Updated onboarding guide with API token creation instructions and required scopes

**Authentication Details:**
- Both methods use HTTP Basic Auth (NOT Bearer tokens)
- API tokens require **Atlassian account email** as username
- App passwords require **Bitbucket username**
- Default for new connections: API Token

**Testing:**
- 26 unit tests added (13 model, 11 API, 2 migration)
- All tests passing
- Backward compatible with existing connections

## Does this close any open issues?
Closes #8520

## Screenshots

**Onboarding Connection Form (API Token - Default)**
<img width="985" height="619" alt="image" src="https://github.com/user-attachments/assets/8272d946-5475-4989-856e-9882f36f8568" />
- Shows deprecation message on the side, warning users about the June 9, 2026 deactivation date.

**New Connection Form (From Config-UI)**
<img width="654" height="688" alt="image" src="https://github.com/user-attachments/assets/a71d7b22-e8bf-4198-a332-bf78b0949e13" />

## Other Information

### Migration Path
- **Existing users**: Connections continue working with App passwords; deprecation warnings guide migration to API tokens
- **New users**: Default to API tokens (recommended method)
- **Zero breaking changes**: All existing connections preserved

### Required API Token Scopes
- `read:account` - Required to view users profiles
- `read:issue:bitbucket` - View your issues
- `read:pipeline:bitbucket` - View your pipelines
- `read:project:bitbucket` - View your projects
- `read:pullrequest:bitbucket` - View your pull requests
- `read:repository:bitbucket` - View your repositories
- `read:runner:bitbucket` - View your workspaces/repositories' runners
- `read:user:bitbucket` - View user info (required for connection test)
- `read:workspace:bitbucket` - View your workspaces

<img width="1636" height="1076" alt="screencapture-id-atlassian-manage-profile-security-api-tokens-2025-10-06-11_13_53" src="https://github.com/user-attachments/assets/a89f5413-b3ee-40fe-936a-cf0c5356e431" />

Commit:
feat(bitbucket): add API token authentication support and deprecate App passwords

- Updated Bitbucket connection model to include `UsesApiToken` field for API token support.
- Modified connection handling in the Bitbucket API to use API tokens.
- Added migration script to update existing connections for backward compatibility.
- Updated UI to reflect changes in authentication method and provide guidance on API token usage.
- Updated documentation to inform users about the deprecation of App passwords.

Commit:
test(bitbucket): add unit tests for API token authentication and connection handling

- Introduced tests for Bitbucket connection API, validating API token and app password authentication methods.
- Added tests for connection sanitization to ensure sensitive data is handled correctly.
- Implemented tests for connection status code handling and deprecation warnings for app passwords.
- Enhanced coverage for connection merging logic and authentication setup.

Addresses #8520

klesh reviewed (commented):
LGTM!
Nice work.
Can you reolve the conflict so I can merge it? Thanks.

Commit:
fix(lint): resolve multiple linting issues

The following issues were resolved:
- SA1006 in `server/api/shared/api_output.go`: Changed `fmt.Errorf` to `errors.Default.New` and removed unused `fmt` import.
- gofmt in `plugins/bitbucket/models/migrationscripts/20251001_add_api_token_auth.go`: Fixed trailing blank line.
- gofmt in `plugins/bitbucket/api/connection_api.go`: Corrected inconsistent tab spacing.
- confusing-results in `server/services/remote/plugin/plugin_impl.go`: Added named return parameters and resolved variable shadowing.
- ST1016 in `plugins/bitbucket/models/connection.go`: Standardized receiver name from `connection` to `bc`.
- S1009 in `plugins/github_graphql/tasks/issue_extractor.go`: Removed redundant nil check.
- superfluous-else in `impls/logruslog/init.go`: Refactored code to eliminate unnecessary else block.
- superfluous-else in `helpers/srvhelper/scope_service_helper.go`: Refactored code to eliminate unnecessary else block.

Fixes #8520

Commit:
Merge branch 'main' into refactor/#8520-bitbucket-api-token
